src/mapblock_mesh.h

```diff
@@ -161,17 +164,69 @@ class MapBlockMesh
 /*
 	This is used because CMeshBuffer::append() is very slow
 */
-struct PreMeshBuffer
+class PreMeshBuffer
 {
+private:
+	bool m_enable_shaders;
+	u32 m_vertex_pitch;
+public:
 	TileSpec tile;
 	std::vector<u16> indices;
 	std::vector<video::S3DVertex> vertices;
+	std::vector<video::S3DVertexTangents> tangent_vertices;
+
+	PreMeshBuffer(bool enable_shaders):
+	m_enable_shaders(enable_shaders)
+	{
+		if (m_enable_shaders)
```

Comment:
It might be a good idea to preallocate some memory here, since we know there are going to be at least a couple hundred vertices.  vertices.reserve(512); or something perhaps